### ISSUE TITLE
Hi, the integration seems to break in HA 23.3.1

### ISSUE THREAD

**Issue by confucius78** [Limych/ha-average#187]
**Environment**
- Home Assistant Core release with the issue: 2023.03.1
- This custom component release with the issue: Latest
- Last working this custom component release (if known): current one was working until now
- Operating environment (Home Assistant/Supervisor/Docker/venv): HA-OS

**Describe the bug**
All values seem to go way off due to some calculation error in the integration. 
![image](https://user-images.githubusercontent.com/21319922/222693973-0745a052-f4dd-414c-98ea-14f2176f26d0.png)

N/A
Add your configs here if any.

**Steps to Reproduce**
Just installed the HA 2023.03.1 update

**Expected behavior**
Kinda stating the obvious, but I'd expect a running average of my sensor that makes sense.

**Debug log**
To be uploaded

Add your logs here.

```

**Additional context**
N/A

**Comment by sshalyminov** [comment]
Same for me
![2023-03-03 155120](https://user-images.githubusercontent.com/25504725/222724884-742b4763-ee3d-48d5-8d0a-aaad6e6d98ff.png)
BTW min and max values of calculated average are exceed min and max of source. For example for this graph source min  and max of source are 5 and 16 but min and max for "average" are 2.2 and 18.83

**Comment by sshalyminov** [comment]
![2023-03-03 181742](https://user-images.githubusercontent.com/25504725/222757615-d4958185-1948-401a-8339-1bebc46ae8ca.png)
Blue line - original value, pink line - calculated "average" value

**Comment by sshalyminov** [comment]
![2023-03-03 183924](https://user-images.githubusercontent.com/25504725/222762580-de9bf2c5-b71e-4f6f-9b8a-a7450ec2626b.png)
blue line - original value, cyan line - calculated by this integration

**Comment by mwright-git** [comment]
Same here!

I'm going to switch to statistics platform as its causing me some major issues. Kept my heating on all night because it claimed the average of two indoor thermometers was -232C!

**Comment by enricarago** [comment]
Same here,  almost all sensors become unavaialable.

**Comment by lmagyar** [comment]
This specific error seems to be an SqlAlchemy query cache bug (again), see: https://github.com/home-assistant/core/issues/89224 and https://github.com/home-assistant/core/issues/89170

Try to restart HA core a few times and there is a good chance to get rid of it until the next restart. Or not.

On the other hand:
- I've added a comment in the forums on how to replace it with the built in integrations, see: https://community.home-assistant.io/t/average-sensor/111674/154
- this integration seems to be abandoned, I've initiated to remove it from the store, see: https://github.com/hacs/integration/issues/3083

**Comment by confucius78** [comment]
Many thanks, I missed that post!

**Comment by lmagyar** [comment]
I've tried to test the PR in [#89224](https://github.com/home-assistant/core/issues/89224) that fixes the query cache issue, but my system currently works without any issue, with or without the PR, even if I restart it 10 times, even if I restore a previous HA core and DB state.

If somebody wants to test the PR also, make sure:
- you have core 2023.3.1 installed
- you have SSH access to HA command line
- you made a full backup

Then you can test it with the following commands:
```
docker exec -it homeassistant /bin/bash
cd /usr/src/homeassistant/
cp homeassistant/components/recorder/history.py /share/
cp homeassistant/components/recorder/statistics.py /share/
curl -L https://github.com/home-assistant/core/files/10903215/0001-pr89254.patch | git apply -v
exit
ha core restart
```

You should see something like this:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 20228  100 20228    0     0  16787      0  0:00:01  0:00:01 --:--:-- 63252
Checking patch homeassistant/components/recorder/history.py...
Checking patch homeassistant/components/recorder/statistics.py...
Hunk #2 succeeded at 645 (offset -4 lines).
Hunk #3 succeeded at 1254 (offset -4 lines).
Hunk #4 succeeded at 1267 (offset -4 lines).
Hunk #5 succeeded at 1280 (offset -4 lines).
Hunk #6 succeeded at 1322 (offset -4 lines).
Hunk #7 succeeded at 1739 (offset -4 lines).
Hunk #8 succeeded at 1918 (offset -4 lines).
Hunk #9 succeeded at 1983 (offset -4 lines).
Hunk #10 succeeded at 2035 (offset -4 lines).
Applied patch homeassistant/components/recorder/history.py cleanly.
Applied patch homeassistant/components/recorder/statistics.py cleanly.
```

If something really goes wrong you can copy back the python files from /share and restart HA core. Or restore a backup.

**Comment by lmagyar** [comment]
That fix is also in 2023.3.2. So no need for command line. :)

Please do some restarts after upgrading to 2023.3.2 to be sure, the problem really disappeared and do not resurface after a restart.

**Comment by PlayFaster** [comment]
Hi There 

I'm still experiencing this issue in HA 2023.3.3. All Temperature Average sensors are unavailable on restart. Can be recovered after a few restarts OR seems to always recover after a full reboot of the HA RaspPi. 

This only affects Temperature based Averages. All others (humidity, Lux, voltage, download speed) are not affected. They are available on every restart and do not seem to exhibit any strange value behaviour either. 

I am seeing the warning pasted below in my HA Log, and I'm wondering if this is related to the issue? 

`WARNING (MainThread) [homeassistant.helpers.frame] Detected integration that uses temperature utility. This is deprecated since 2022.10 and will stop working in Home Assistant 2023.4, it should be updated to use unit_conversion.TemperatureConverter instead. Please report issue to the custom integration author for average using this method at custom_components/average/sensor.py, line 258: temperature = convert_temperature(float(temperature), entity_unit, ha_unit) 
`

**Comment by lmagyar** [comment]
The database related issue is fixed in sqlalchemy and will be part of 2023.3.4, fingers crossed https://github.com/home-assistant/core/issues/89489#issuecomment-1466640880

The deprecation problem I think will never be fixed in this integration. :( I've moved on and replaced it with the built-in sensors. An idea in case of temperature averages: https://community.home-assistant.io/t/average-sensor/111674/153

**Comment by PlayFaster** [comment]
Hi Imagyar 

Thanks for this. I have this week also made the move to the built-in-sesnors. The Average integration certainly has it's advantages, especially when it comes to averaging several sensors over time, but it definately has it's issues as well! 

Thanks
